java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -332,23 +348,31 @@ public FlightInfo getCrossReference(String pkCatalog, String pkSchema, String pk
         getOptions());
   }
 
+  /**
+   * Config options shared by both the {@link ArrowFlightSqlClientHandler.Builder}
+   * and the {@link ConnectionFactory}.
+   */
+  static class Config {
```

Comment:
Move the builder config into this data class, so it can be re-used later from the factory without breaking the builder API.